docx_editor/ooxml/unpack.py

```
@@ -99,6 +120,9 @@ def unpack_document(input_file: str | Path, output_dir: str | Path) -> str:
     # Reject a symlinked destination so extractall cannot write through it.
     if output_path.is_symlink():
         raise InvalidDocumentError(f"Output directory is a symlink: {output_dir}", path=output_path)
+    ancestor = _nearest_existing(output_path)
+    if ancestor.is_symlink():
+        raise InvalidDocumentError(f"Output directory would be created through a symlink: {ancestor}", path=ancestor)
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)